docs/HowToFAQ/Errors-Resolution.md

```diff
@@ -519,15 +573,67 @@ considered ghost. And then the statement can contain no substatements that are f
 `return` and `yield` stastements are never ghost, so they cannot appear in a statement whose guarding
 value is ghost.
 
-## **cannot assign to _var_ in a ghost context**
+## **Error: cannot assign to _var_ in a ghost context** {#r_no_assign_to_var_in_ghost}
 
-<!-- TODO -->
+```dafny
+method m(ghost c: int) 
+{
+  var x := 7;
+  if (c == 1) { x :| x < 8; }
+}
+```
 
-## **_var_ cannot be assigned a value that depends on a ghost**
+No changes to non-ghost variables may occur in ghost contexts.
+Ghost context can be implicit, such as the branches of an if-statement
+whose condition is a ghost expression.
 
-<!-- TODO -->
+## **Error: non-ghost _var_ cannot be assigned a value that depends on a ghost** {#r_no_assign_ghost_to_var}
 
-## **Error: in _proof_, calls are allowed only to lemmas**
+```dafny
+method m(ghost c: int) 
+{
+  var x := 8;
+  x :| x < c;
+}
+```
+
+In a assign-such-that statement, the LHS gets some value that satisfies the boolean expression on the RHS.
+If the LHS is non-ghost, then the RHS must be non-ghost, because non-ghost code may not depend on
+ghost code.
+
+## **Error: assumption variable must be of type 'bool'** {#r_assumption_var_must_be_bool}
+
+```dafny
+method m() {
+  ghost var {:assumption} b: int;
+}
+```
+
+Variables marked with `{:assumption} must have bool type.
+See [the reference manual](#../DafnyRef/DafnyRef#sec-assumption) for more detail on the use of this attribute.
+
+
+## **Error: assumption variable must be ghost** {#r_assumption_var_must_be_ghost}
```

Comment:
```suggestion
## **Error: assumption variables must be ghost** {#r_assumption_var_must_be_ghost}
```

Or is the typo in the actual error message?

Reply:
The typo is in the error message, some of which I have been fixing (if they don't involve fixing too many tests).
Actually in this form it is just cryptic: could be 'this assumption variable must be ghost'

Reply:
Agreed and I almost skipped this one since error messages are often this cryptic/abstract, but really that's no excuse and they SHOULD be more readable :)